nibabel/testing/__init__.py

```diff
@@ -56,22 +52,50 @@ def assert_allclose_safely(a, b, match_nans=True):
     assert_true(np.allclose(a, b))
 
 
-class suppress_warnings(catch_warnings):
-    """ Version of ``catch_warnings`` class that suppresses warnings
-    """
-    def __enter__(self):
-        res = super(suppress_warnings, self).__enter__()
-        simplefilter('ignore')
-        return res
+class clear_and_catch_warnings(warnings.catch_warnings):
```

Comment:
This is a copy of code from https://github.com/numpy/numpy/pull/5682/files . I did NOT copy over the tests.